### PR TITLE
calibre: build with py3 by default

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

python 2 is dead and calibre has been supporting py3 for a while now.

In case people still need py2 due to external plugins not supporting py3
yet, we have calibre-py2.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by peterhoeg** [comment]
Cc @domenkozar @pSub @AndersonTorres

**Comment by peterhoeg** [comment]
We probably want this in 20.09 @jonringer due to the py2 -> py3 move.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -19691,7 +19691,11 @@ in
 
   calculix = callPackage ../applications/science/math/calculix {};
 
-  calibre = libsForQt5.callPackage ../applications/misc/calibre { };
+  calibre-py2 = libsForQt5.callPackage ../applications/misc/calibre { pythonPackages = python2Packages; };
+
+  calibre-py3 = libsForQt5.callPackage ../applications/misc/calibre { pythonPackages = python3Packages; };
+
+  calibre = calibre-py3;
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:19698]: I believe it can make a difference with overlays
```suggestion
  calibre = self.calibre-py3;
```

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (peterhoeg) [reply pkgs/top-level/all-packages.nix]: I'll take your word for it. Fixed! Bumped the version while at it. Works here.

Comment (peterhoeg) [reply pkgs/top-level/all-packages.nix]: So grahamcofborg begs to differ:

```
nix-env did not evaluate cleanly:
 ["trace: It seems that you are using a patched Nixpkgs that references the self", "variable in pkgs/top-level/all-packages.nix. This variable was incorrectly", "named, so its usage needs attention. Please use pkgs for packages or super", "for functions.", "trace: It seems that you are using a patched Nixpkgs that references the self", "variable in pkgs/top-level/all-packages.nix. This variable was incorrectly", "named, so its usage needs attention. Please use pkgs for packages or super", "for functions.", "trace: It seems that you are using a patched Nixpkgs that references the self", "variable in pkgs/top-level/all-packages.nix. This variable was incorrectly", "named, so its usage needs attention. Please use pkgs for packages or super", "for functions.", "trace: It seems that you are using a patched Nixpkgs that references the self", "variable in pkgs/top-level/all-packages.nix. This variable was incorrectly", "named, so its usage needs attention. Please use pkgs for packages or super", "for functions."]
```

Comment (jonringer) [reply pkgs/top-level/all-packages.nix]: interesting, not what I was expecting, as this works in python packages.

**Comment by obadz** [comment]
Can no longer use calibre without reverting 
3690d1d :

```
$ ebook-viewer 
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.

Aborted (core dumped)
```

(Same error for both `py2` and `-py3`)

Repro: `nix-shell --pure -p calibre --command ebook-viewer`

**Comment by mtreca** [comment]
See #98067

**Comment by peterhoeg** [comment]
Should be fixed by https://github.com/NixOS/nixpkgs/pull/98197.

**Pull request merged**